### PR TITLE
NPE in CodeStream.retrieveLocalType()

### PR DESCRIPTION
**Pull request by iloveeclipse**
Fixes https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2434

TODO: regression test

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -35,7 +35,7 @@ pipeline {
 						-Dmaven.test.failure.ignore=true \
 						-Dcompare-version-with-baselines.skip=false \
 						-Djava.io.tmpdir=$WORKSPACE/tmp -Dproject.build.sourceEncoding=UTF-8 \
-						-Dtycho.surefire.argLine="--add-modules ALL-SYSTEM -Dcompliance=1.8,11,17,21,22 -Djdt.performance.asserts=disabled" \
+						-Dtycho.surefire.argLine="--add-modules ALL-SYSTEM -Dcompliance=1.5,1.8,11,17,21,22 -Djdt.performance.asserts=disabled" \
```
Comment (iloveeclipse) [inline Jenkinsfile:38]: Note I've added `1.5` JLS temporarily to see if any test fail on 1.5

**Comment by srikanth-sankaran** [comment]
There are 4-5 enum tests that were noticed to fail locally. This is in my follow up tasks list.

**Comment by srikanth-sankaran** [comment]
See https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2234#issuecomment-2100738652

**Comment by srikanth-sankaran** [comment]
> See https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2234#issuecomment-2100738652

These may in fact go away with the suggested fix. Let us see.

**Comment by iloveeclipse** [comment]
All tests are green. 
Closing this now (as it was only opened to see test results), @srikanth-sankaran will provide the right PR with regression test.

**Pull request closed without merging**